src/scandipwa/app/component/GoogleTagManager/events/AddToCart.event.js

```diff
@@ -48,6 +52,8 @@ class AddToCartEvent extends BaseEvent {
      * Handle product add to cart
      */
     handler(product, quantity, isItem, isGrouped, isFromCart, massAddAction) {
+        if (!isEventEnabled(EVENT_ADD_TO_CART)) return;
```

Comment:
I think this, can be deduplicated as well by using the plugin system, wrapping around all these classes' `handler` methods and validating whether the event is activated to decide whether to call the handler. Although this might be redundant if we remove the event dispatches and activation.